lib/nmatrix/shortcuts.rb

```diff
@@ -465,7 +551,54 @@ def random(shape, opts={})
       NMatrix.new(shape, random_values, {:dtype => :float64, :stype => :dense}.merge(opts))
     end
     alias :rand :random
-
+    
+    # 
+    #  call-seq:
+    #    magic(shape) -> NMatrix
+    #
+    #  The parameter is the dimension of the matrix.
+    #
+    #  Creates a +:dense+ NMatrix with the following properties:
+    #    - An arrangement of the numbers from 1 to n^2 (n-squared) in the matrix, with each number occurring exactly once.
+    #    - The sum of the entries of any row, any column, or any main diagonal is the same.
+    #    - This sum must be n(n^2+1)/2.
+    #   
+    #  See: http://www.mathworks.com/help/matlab/ref/magic.html
+    # 
+    #  * *Arguments* :
+    #   - +shape+ -> Array (or integer for square matrix) specifying the dimensions.
+    #  * *Returns* :
+    #   - NMatrix with the above given properties.
+    #
+    #  Examples:
+    #       
+    #    NMatrix.magic(3) # => [  [4, 9, 2]   [3, 5, 7]   [8, 1, 6] ]
+    #    
+    #    NMatrix.magic(4) # => [  [ 1, 15, 14,  4]
+    #                             [12,  6,  7,  9]
+    #                             [ 8, 10, 11,  5]
+    #                             [13,  3,  2, 16] ]
+    #                             
+    #    NMatrix.magic(6) # => [  [31,  9,  2, 22, 27, 20]
+    #                             [ 3, 32,  7, 21, 23, 25]
+    #                             [35,  1,  6, 26, 19, 24]
+    #                             [ 4, 36, 29, 13, 18, 11]
+    #                             [30,  5, 34, 12, 14, 16]
+    #                             [ 8, 28, 33, 17, 10, 15] ]
+    #
+    def magic(shape)
+      if shape==2 then raise(ArgumentError, "shape of two is not allowed") end
```

Comment:
A better way is `raise(ArgumentError, "shape of two is not allowed") if  shape == 2`